woocommerce/woocommerce
# Product Block Editor: move logic to add new attributes to modal component

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

* Move the logic to create a new attribute from the input component to the attribute modal
* Tweak types to be more consistent. Remove multiple type for the product attribute
* Clean unused code, for instance, `getProductAttributeObject`.
* Remove tests





Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Test adding blocks attributes work as usual (copied from https://github.com/woocommerce/woocommerce/pull/46655)

1. Enable the new Product Editor
2. Create a new product
3. Go to the `Variations` section
4. Click on the `Add options` button to add new attributes

<img width="754" alt="image" src="https://github.com/woocommerce/woocommerce/assets/77539/ca220878-9d0c-44a2-8f6b-8c15faf03c04">

6. Add a few attributes (I added `Size`, `Color` and `Shape`)

<img width="820" alt="image" src="https://github.com/woocommerce/woocommerce/assets/77539/d668d3b0-ec87-4522-9f50-57d807d5fa53">

8. Try to add attribute options from the same modal
9. Confirm that, as long as you start adding new attribute options, the select dropdown shows fewer attributes

<img width="827" alt="image" src="https://github.com/woocommerce/woocommerce/assets/77539/5c06c5e7-5be9-4565-811e-2ccb5ff4f580">


11. Cancel (to keep testing)
12. Add an attribute from the `Organization` section (I added the `Color` attribute)


<img width="743" alt="Screenshot 2024-04-16 at 18 11 02" src="https://github.com/woocommerce/woocommerce/assets/77539/5aa29b56-8db9-495b-9c00-dd6e22082140">

13. Save
14. Go back to the `Variations` section
15. Start to add a new attribute
16. Confirm that the `Color` item is disabled


<img width="842" alt="Screenshot 2024-04-16 at 18 13 06" src="https://github.com/woocommerce/woocommerce/assets/77539/c89fc69c-9a37-4dc6-8482-2cd71acd9753">



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
